src/main/java/filter_post/interface_adapters/FilterPostViewModel.java

```diff
@@ -0,0 +1,76 @@
+package filter_post.interface_adapters;
+
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyChangeSupport;
+
+public class FilterPostViewModel {
+    // Using parallel arrays to store the post data.
+    private String[] titles;
+    private int[] ids;
+    private String[] descriptions;
+    private final PropertyChangeSupport observable;
+
+    /**
+     * Initialize the view model for the filter post use case.
+     * @param titles        A list of titles of all the posts.
+     * @param ids           A list of ids of all the posts.
+     * @param descriptions  A list of descriptions of all the posts.
+     */
+    public FilterPostViewModel(String[] titles, int[] ids, String[] descriptions) {
+        this.titles = titles;
+        this.ids = ids;
+        this.descriptions = descriptions;
+        this.observable = new PropertyChangeSupport(this);
+    }
+
+    /**
+     * Add a new observer to observe changes to this class.
+     * @param observer The observer to be observing this observable.
+     */
+    public void addObserver(PropertyChangeListener observer) {
```

Comment:
This is more of a note to self but we need to implement the observer that is passed here in the view post use case